app/components/welcome-message.js

```diff
@@ -0,0 +1,26 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import { notEmpty } from '@ember/object/computed';
+
+export default Component.extend({
+  session: service(),
+
+  text: computed('session.currentUser.{email_verified,has_tokens}', function() {
+    const user = this.get('session.currentUser');
+    if (!user || (user.email_verified && user.has_tokens)) return '';
+
+    const textArray = [
+      !user.email_verified && 'verify your email address',
+      !user.email_verified && !user.has_tokens && ' and ',
+      !user.has_tokens && 'create an API token',
+      '!',
+    ].filter(e => !!e);
+
+    return textArray.join('');
+  }),
```

Comment:
I think I would prefer it if we moved that code into the index controller and not instantiate the component at all if we don't have anything to display

Reply:
This logic has nothing in common with the index controller. This is common arch mistake in all MVC frameworks. Trying to move business logic inside of controllers fail by design.